data/src/normalize.py

```diff
@@ -0,0 +1,39 @@
+import argparse
+
+from tempfile import TemporaryFile
+from unicodedata import normalize
+
+'''
+normalize.py
+
+Takes a file and normalizes it "in place." In order to avoid the issues of
+reading and writing to the same file at the same time, this script puts the 
+normalized version of the file argument in a tempfile, then uses that tempfile
+to rewrite the original file.
+'''
+
+def main(args: argparse.Namespace) -> None:
+	tf = TemporaryFile(mode='w+')
+
+	with open(args.file, 'r') as rf:
+		for line in rf:
+			tf.write(normalize(args.norm, line))
```

Comment:
Prefer `print` to `write`. (it has an option to not add newlines if that's what you want.)

Reply:
Is there a practical reason for this, or is it just a stylistic thing? Just wondering.

Reply:
I suppose it's stylistic at this point: in the 16-odd years I've been doing this, I've almost never seen `write` used.